WordPress/Classes/ViewRelated/Blog/QuickStartPromptViewController.swift

```diff
@@ -0,0 +1,162 @@
+        promptDescriptionLabel.textColor = .textSubtle
+
+        showMeAroundButton.isPrimary = true
+        noThanksButton.isPrimary = false
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        setupSiteInfoViews()
+        setupPromptInfoViews()
+        setupButtons()
+    }
+
+    private func setupSiteInfoViews() {
+        siteIconView.downloadSiteIcon(for: blog)
+
+        let displayURL = blog.displayURL as String? ?? ""
+        if let name = blog.settings?.name?.nonEmptyString() {
+            siteTitleLabel.text = name
+            siteDescriptionLabel.text = displayURL
+        } else {
+            siteTitleLabel.text = displayURL
+            siteDescriptionLabel.text = nil
+        }
+    }
+
+    private func setupPromptInfoViews() {
+        promptTitleLabel.text = Strings.promptTitle
+        promptDescriptionLabel.text = Strings.promptDescription
+    }
+
+    private func setupButtons() {
+        showMeAroundButton.setTitle(Strings.showMeAroundButtonTitle, for: .normal)
+        noThanksButton.setTitle(Strings.noThanksButtonTitle, for: .normal)
+    }
+
+    // MARK: - IBAction
+
+    @IBAction private func showMeAroundButtonTapped(_ sender: Any) {
+        dismissPrompt { [weak self] in
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + Constants.quickStartDelay) {
+                guard let self = self else {
+                    return
+                }
+                QuickStartTourGuide.shared.setup(for: self.blog)
+            }
+        }
+    }
+
+    @IBAction private func noThanksButtonTapped(_ sender: Any) {
+        UserDefaults.standard.setQuickStartWasDismissed(true, for: blog)
+        dismissPrompt()
+    }
+
+    private func dismissPrompt(completion: (() -> Void)? = nil) {
+        onDismiss?(blog)
+        navigationController?.dismiss(animated: true, completion: completion)
+    }
+}
+
+extension QuickStartPromptViewController {
+
+    private enum Strings {
+        static let promptTitle = NSLocalizedString("Want a little help managing this site with the app?", comment: "Title for a prompt asking if users want to try out the quick start checklist.")
+        static let promptDescription = NSLocalizedString("Learn the basics with a quick walk through.", comment: "Description for a prompt asking if users want to try out the quick start checklist.")
+        static let showMeAroundButtonTitle = NSLocalizedString("Show me around", comment: "Button title. When tapped, the quick start checklist will be shown.")
+        static let noThanksButtonTitle = NSLocalizedString("No thanks", comment: "Button title. When tapped, the quick start checklist will not be shown, and the prompt will be dismissed.")
+    }
+
+    private enum Constants {
+        static let quickStartDelay: DispatchTimeInterval = DispatchTimeInterval.milliseconds(1000)
+    }
+}
+
+extension UserDefaults {
+
+    func quickStartWasDismissed(for blog: Blog) -> Bool {
```

Comment:
I have mixed feelings about extending `UserDefaults` to account for quick start stuff.

Do you think it would be too complicated to create a new entity responsible for that and inject where is needed?

Reply:
@leandroalonso 👋 

> Do you think it would be too complicated to create a new entity responsible for that and inject where is needed?

Do you mean like create a QuickStartDefaultsManager / Helper that handles everything? I feel that may be a bit overengineered, but I'm curious why you have mixed feelings about extending `UserDefaults` -- maybe I'm overlooking something -- can you elaborate?

Reply:
I agree with @leandroalonso here.

The issue with accessing singletons (like `UserDefaults`) directly throughout the code is that the code becomes much harder to unit test because [the method calls are not deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm), requiring you to mock the singleton (`UserDefaults.standard`) which isn't great and sometimes can bring several other complications if running several tests in parallel using the same mocked singleton.

An alternative to this can be dependency injection, but most of the times you don't really need a new class.

As an example, if I'm implementing `SomeClass` and I need it to use a singleton, I try to use a pattern like this:

```swift
    class SomeClass {
        private let someKey = "someKey"
        private let userDefaults: UserDefaults

        init(userDefaults: UserDefaults = .standard) {
            self.userDefaults = userDefaults
        }

        // Mark - User Defaults Storage

        func getSomeDate() -> Date? {
            userDefaults.object(forKey: birthday) as? Date
        }

        func setSomeDate(_ date: Date) {
            userDefaults.set(date, forKey: someKey)
        }
    }
```

This means that when I'm unit-testing this class I can do:

```swift
let someObj = SomeClass(userDefaults: myMockUserDefaults)

// ...test away
```

DISCLAIMER: I'm being a bit purist here and I've failed to do this myself more than once.  I still think it's worth it to try and use this approach.

Reply:
Fixed! 

fe54651
94ba763 
c7c3c26